### DISCUSSION TITLE
Expression depending on the dimension of the case

### DISCUSSION THREAD

**Discussion by thomas-saigre** [feelpp/feelpp#2031] [Q&A]
In the doc we can find symbols associated to specific toolboxes (e.g. `heat_T`), but is there a list of all the default symbols available in Feel ? For instance `x`, `y`, `t`...

In particular, can we access the dimension of the case in expression ? For instance, I'm trying to make one looking like :
```
"{0,0}*(dim==2) + {0,0,0}*(dim==3):dim"
```
which is equals to `{0,0}` when we run a 2D case, and `{0,0,0}` for 3D.

**Comment by prudhomm** [comment]
you have this page https://docs.feelpp.org/user/latest/using/expr.html
the reserved symbols are : `x,y,z,t,nx,ny,nz` and a few other ones. I should probably update the documentation.

`nx,ny,nz` are defined only when the expression is evaluated on facets

**Reply by thomas-saigre** [reply]
is the dimension accessible ?
